plenum/test/pool_transactions/test_get_txn_request.py

```diff
@@ -1,129 +1,129 @@
-from plenum.common.constants import POOL_LEDGER_ID, DOMAIN_LEDGER_ID, CLIENT_BLACKLISTER_SUFFIX, \
-    NODE_BLACKLISTER_SUFFIX, NODE_PRIMARY_STORAGE_SUFFIX, HS_FILE, HS_LEVELDB, TXN_TYPE, LedgerState, LEDGER_STATUS, \
-    CLIENT_STACK_SUFFIX, PRIMARY_SELECTION_PREFIX, VIEW_CHANGE_PREFIX, OP_FIELD_NAME, CATCH_UP_PREFIX, NYM, \
-    POOL_TXN_TYPES, GET_TXN, DATA, MONITORING_PREFIX, TXN_TIME, VERKEY, TARGET_NYM, ROLE, STEWARD, TRUSTEE, ALIAS, \
-    NODE_IP, CURRENT_PROTOCOL_VERSION
+import json
+
+import pytest
+
+from indy.ledger import build_get_txn_request
 from random import randint
 
-from plenum.common.types import f
-from plenum.test.pool_transactions.helper import sendAddNewClient
+from plenum.common.constants import INVALID_LEDGER_ID, INVALID_SEQ_NO, DATA
+from plenum.common.exceptions import RejectError
+from plenum.test.pool_transactions.helper import sdk_sign_and_send_prepared_request, \
+    prepare_nym_request
 from stp_core.loop.eventually import eventually
-from plenum.test import waits
-from plenum.test.helper import check_sufficient_replies_received, \
-    waitReqNackWithReason
-from plenum.common.util import getMaxFailures
-from plenum.common.request import Request
+from plenum.test.helper import sdk_get_and_check_replies
+from plenum.common.util import getMaxFailures, randomString
 
 c_delay = 10
 fValue = getMaxFailures(4)
 
 
 def test_get_txn_for_invalid_ledger_id(looper, txnPoolNodeSet,
-                                       steward1, stewardWallet):
-    op = {
-        TXN_TYPE: GET_TXN,
-        f.LEDGER_ID.nm: 5908,
-        DATA: 1
-    }
-    req = Request(identifier=stewardWallet.defaultId,
-                  operation=op, reqId=Request.gen_req_id(),
-                  protocolVersion=CURRENT_PROTOCOL_VERSION)
-    steward1.submitReqs(req)
-    for node in txnPoolNodeSet:
-        waitReqNackWithReason(looper, steward1, 'expected one of',
-                              node.clientstack.name)
+                                       sdk_wallet_steward,
+                                       sdk_pool_handle):
+    _, steward_did = sdk_wallet_steward
+    request = looper.loop.run_until_complete(
+        build_get_txn_request(steward_did, 1))
```

Comment:
I would prefer not to use raw sdk functions - create wrapper in conftest/helper, add test to sdk_tests

Reply:
Done